src/pudl/output/ferc1.py

```diff
@@ -3053,36 +3049,69 @@ def out_ferc1__yearly_rate_base(
             disaggregate_null_or_total_tag,
             tag_col="tags_aggregatable_utility_type",
         )
+        # bc we just disaggreated the utility type tag, there are some
+        # totals in the utilty_type column. even though these two columns
+        # do not need to stay consistent, it feels cleaner to keep
+        # these consistent for these dissagregated records
+        .assign(
+            utility_type=lambda x: np.where(
+                x.is_disaggregated_tags_aggregatable_utility_type,
+                x.tags_aggregatable_utility_type,
+                x.utility_type,
+            )
+        )
         .pipe(
             disaggregate_null_or_total_tag,
             tag_col="tags_in_rate_base",
         )
     )
 
     in_rate_base_df = rate_base_df[rate_base_df.tags_in_rate_base == "yes"]
-    return in_rate_base_df.dropna(subset=["ending_balance"])
-
-
-@asset_check(
-    asset="out_ferc1__yearly_rate_base",
-    additional_ins={"tags_df": AssetIn("_out_ferc1__detailed_tags")},
-    blocking=True,
-)
-def check_tag_propagation(out_ferc1__yearly_rate_base, tags_df: pd.DataFrame):
-    """Check propagated tags compared to manually compiled tags."""
     for tag in ["in_rate_base", "aggregatable_utility_type"]:
         # note: we need the `tags_in_rate_base` column for these checks
         check_tag_propagation_compared_to_compiled_tags(
-            out_ferc1__yearly_rate_base, tag, tags_df
+            in_rate_base_df, tag, _out_ferc1__detailed_tags
         )
-    return AssetCheckResult(passed=True)
+        check_for_correction_xbrl_factoids_with_tag(in_rate_base_df, tag)
+    return in_rate_base_df.dropna(subset=["ending_balance"])
 
 
 @asset_check(asset="out_ferc1__yearly_rate_base", blocking=True)
-def check_for_correction_tags(out_ferc1__yearly_rate_base):
-    """Check if any correction records have key tags."""
-    for tag in ["in_rate_base", "aggregatable_utility_type"]:
-        check_for_correction_xbrl_factoids_with_tag(out_ferc1__yearly_rate_base, tag)
+def check_pks(df):
+    """Check the primary keys of this table.
+
+    We do this as an asset check instead of actually setting the them as primary keys
+    because there are many expected nulls in these columns.
+    """
+    idx = [
+        "report_year",
+        "utility_id_ferc1",
+        "xbrl_factoid",
+        "table_name",
+        "utility_type",
+        "plant_function",
+        "plant_status",
+        # this needs to be in here bc xbrl_factoid==utility_plant_net_correction
+        # had correcitons w/ total and sub-dim utility types
```

Comment:
What is "sub-dim"?

Reply:
sub-dimension. I will write this out. there is a ton of language and documentation in the transform module about what that is.